### PR TITLE
duplicate op context methods on asset context 

### PR DESCRIPTION
**Pull request by jamiedemaria**
## Summary & Motivation
Duplicates all methods/properties on `OpExecutionContext` to `AssetExecutionContext` as pass throughs that just call the underlying `OpExecutionContext`. This is largely to help with bookkeeping as we deprecate op context methods. The idea being that having a copy of all of the methods on `AssetExecutionContext` will give us a clearer idea which methods are yet to be deprecated. 

This also introduces a decorator to copy the doc string from the op context fn to the asset context fn

### CODE REVIEW

**Comment by jamiedemaria** [comment]
Current dependencies on/for this PR:
* master
  * **PR #17554** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/17554?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #18573** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/18573?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈
      * **PR #17339** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/17339?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
        * **PR #18549** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/18549?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
          * **PR #18543** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/18543?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
            * **PR #18553** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/18553?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
        * **PR #17972** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/17972?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
          * **PR #18044** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/18044?utm_source=stack-comment-icon" target="_blank">Graphite</a> 

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/dagster-io/dagster/18573?utm_source=stack-comment).

File: python_modules/dagster/dagster/_core/execution/context/compute.py [hunk 1]
```diff
@@ -1354,9 +1363,18 @@ def get() -> "OpExecutionContext":
         return ctx.get_op_execution_context()
 
 
+def _copy_docs_from_op_execution_context(obj: T_Annotatable) -> T_Annotatable:
```
Comment (jamiedemaria) [inline python_modules/dagster/dagster/_core/execution/context/compute.py:1366]: is there any use in this being a more general purpose decorator? If so I could move it to `_annotations.py`

**Comment by smackesey** [comment]
>This also introduces a decorator to copy the doc string from the op context fn to the asset context fn

🙏

File: python_modules/dagster/dagster/_core/execution/context/compute.py (hunk 1 above)
Comment (smackesey) [reply python_modules/dagster/dagster/_core/execution/context/compute.py:1366]: I don't think so-- and strictly speaking it isn't an annotation method (at least as I intended the term). I think to simplify I would remove `_get_annotation_target` and `isfunction` here, pretty sure you don't need them-- though I would check the docs rendering or maybe just add a line to your test that ensures docstrings match.

File: python_modules/dagster/dagster/_core/execution/context/compute.py [hunk 2]
```diff
@@ -1365,9 +1383,365 @@ def get() -> "AssetExecutionContext":
...
+    @property
+    @_copy_docs_from_op_execution_context
+    def typed_event_stream_error_message(self) -> Optional[str]:
+        return self.op_execution_context.typed_event_stream_error_message
+
+    @_copy_docs_from_op_execution_context
+    def set_requires_typed_event_stream(self, *, error_message: Optional[str] = None) -> None:
+        self.op_execution_context.set_requires_typed_event_stream(error_message=error_message)
+
     @cached_method
```
Comment (smackesey) [inline python_modules/dagster/dagster/_core/execution/context/compute.py:1742]: Don't need `cached_method` anymore right

File: python_modules/dagster/dagster/_core/execution/context/compute.py (hunk 1 above)
Comment (jamiedemaria) [reply python_modules/dagster/dagster/_core/execution/context/compute.py]: ok good to know. went ahead and added the test!

**Review by alangenfeld — APPROVED** [review]

**Pull request merged**